### PR TITLE
Fixed - Find Friend By Name PR [READY_FOR_REVIEW]

### PR DESCRIPTION
**Pull request by jpgcz**
# GreenCityUser | fixed - Find Friend By Name PR [NEED_REVIEW]

## Summary Of Changes :fire:
The search result didn't work when searching for friend by name and it was because the DB changed how it worked

## Issue Link 🔗 
[Card #4852](https://github.com/ita-social-projects/GreenCity/issues/4852)

## Changed
- Changed Query on UserRepo
- The query searched the status by a integer before and now search by a string because the data in DB is different

## How to test
This could be tested with Swagger _(Needs to have GreenCityChat application running)_
1. Open the link once you ran the application [Here](http://localhost:8060/swagger-ui.html#/)
2. Look for user-controller and open it
3. Click on the GET method /user/findFriendByName and test it! ✔️ 

# PR Checklist Forms ✔️ 
- [x] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] [Not needed] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-XXXX: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [x] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x]  All files reviewed before sending to reviewers

### CODE REVIEW

File: service/src/main/java/greencity/service/UserServiceImpl.java [hunk 1]
```diff
@@ -1134,8 +1134,6 @@ public PageableDto<UserAllFriendsDto> findFriendByName(String name, Pageable pag
             new TypeToken<List<UserAllFriendsDto>>() {
             }.getType());
 
-        friendDtos.stream().forEach(f -> f.setFriendsChatDto(restClient.chatBetweenTwo(f.getId(), id)));
-
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/UserServiceImpl.java]: Why did you remove it?
Should it also be removed from `findAllUsersFriends` method?

**Review by RomaMocherniuk — APPROVED** [review]

**Review by JoseCaste — APPROVED** [review]

**Review by juseti — APPROVED** [review]

**Review by LiliaMokhnatska — APPROVED** [review]
